### PR TITLE
Add dependency-check plugin to identify security vulnerabilities

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

See https://github.com/gradle/gradle/issues/3217

### CODE REVIEW

**Comment by bmuschko** [comment]
@oehme I think this change is fairly check to review and integrate with the benefit of having better visibility into our vulnerabilities. Do we want to press on with this?

**Comment by bmuschko** [comment]
@oehme As discussed I tried hooking up the dependency checks to the sanity check. It adds about 2 1/2 minutes of overhead for every commit. I don't think we'll want to have such a hit on every commit. I'd still advocate for a different CI job alongside the smoke tests.

Also, the CVSS threshold for failing the build is relatively high at the moment. Over time we should work on bringing down the number.

File: gradle/dependenciesCheck.gradle [hunk 1]
```diff
@@ -21,8 +21,20 @@ buildscript {
...
+    apply plugin: org.owasp.dependencycheck.gradle.DependencyCheckPlugin
+
+    dependencyCheck {
+        failBuildOnCVSS = 8 // 10 is the maximum
+        skipConfigurations = ['jmh']
+    }
+}
+
+def projectsIncludedInDistribution() {
+    def excluded = [':buildScanPerformance', ':distributions', ':docs', ':integTest', ':internalIntegTesting', ':internalPerformanceTesting', ':internalTesting', ':performance', ':smokeTest', ':soak']
```
Comment (oehme) [inline gradle/dependenciesCheck.gradle]: ⭕️ Cast to `Set` to make the `contains()` further down efficient.

File: gradle/dependenciesCheck.gradle [hunk 2]
```diff
@@ -21,8 +21,20 @@ buildscript {
...
+        classpath 'org.owasp:dependency-check-gradle:3.0.2'
     }
 }
 
-apply plugin: org.owasp.dependencycheck.gradle.DependencyCheckPlugin
+configure(projectsIncludedInDistribution()) {
+    apply plugin: org.owasp.dependencycheck.gradle.DependencyCheckPlugin
+
+    dependencyCheck {
+        failBuildOnCVSS = 8 // 10 is the maximum
```
Comment (oehme) [inline gradle/dependenciesCheck.gradle]: ⭕️ "10 is the maximum" doesn't really tell me what this means. Do the numbers have specific meanings or are they arbitrarily assigned?

Comment (bmuschko) [reply gradle/dependenciesCheck.gradle]: The Common Vulnerability Scoring System (CVSS) uses several aspects of exploit-ability and impact to determine a base score. Each vulnerability in the NVD has a base score attached, and anything above 4.0 is considered Medium or High severity ([CVSS overview](https://nvd.nist.gov/vuln-metrics/cvss)).

It's a calculated value. The value 10 indicates Critical severity. So with the value 8 we didn't score so well. In a follow up issue we should try to get down the number.

File: gradle/dependenciesCheck.gradle (hunk 1 above)
Comment (bmuschko) [reply gradle/dependenciesCheck.gradle]: done

**Comment by bmuschko** [comment]
> I agree that adding >2min to the sanity check is not a good idea. Let's add it as part of the branch build accept.

I am going to open a separate PR for adding the new job to the TeamCity configuration.

**Comment by bmuschko** [comment]
@oehme Are we ready to merge this?

**Comment by oehme** [comment]
Yes, we are

**Pull request merged**